.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,111 @@
+name: Docker Build and Push Check
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'Branch or commit ID'
+        required: false
+        type: string
+
+jobs:
+  build-and-push-dockerfile:
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        include:
+          - platform: linux/amd64
+            tag: linux-amd64
+            os: ubuntu-latest
+          - platform: linux/arm64
+            tag: linux-arm64
+            os: ubuntu-24.04-arm
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.ref }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract short SHA
+        id: vars
+        run: |
+          if [[ -n "${{ github.event.inputs.ref }}" ]]; then
+            echo "SHORT_SHA=${{ github.event.inputs.ref }}" >> $GITHUB_ENV
+          else
+            echo "SHORT_SHA=${GITHUB_SHA::7}" >> $GITHUB_ENV
+          fi
+
+      - name: Extract metadata for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository }}
+          tags: |
+            type=ref,event=branch,separator=_
+            type=raw,value=${{ env.SHORT_SHA }}
+
+      - name: Build Docker image (Dockerfile)
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: Dockerfile
+          platforms: ${{ matrix.platform }}
+          tags: ghcr.io/${{ github.repository }}:${{ env.SHORT_SHA }}-${{ matrix.tag }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+        env:
+          CARGO_INCREMENTAL: ${{ github.event_name == 'pull_request' && '1' || '0' }}
```

Comment:
This pipeline is not at all triggered by the `pull_request` event, so we won't need to test for the `pull_request` event?